### DISCUSSION TITLE
RGBW not working as expected (Sunricher 4 in 1 Universal ZigBee LED Controller)

### DISCUSSION THREAD

**Discussion by dawiinci** [Koenkk/zigbee2mqtt#13220] [Devices]
I want to use the white channel at 100% and fine tune the color with RGB. But it seems that I can only change the color temperature **or** the RGB. That means I will never get 100% power and full white. This is also visible as the light seems much dimmer than before with another RGB controller.

Am I missing something?

The switches are set for RGBW according the manual (black square is where the switch is). I also connected RGBW and + to the stripe like before. I have two lights in this configuration. The effect is the same, so hardware failure seems unlikely.

https://www.sunricher.com/media/resources/manual/SR-ZG1029-5C%20V2%20instruction.pdf
https://www.sunricher.com/dim-cct-rgbw-rgb-cct-4-in-1-constant-voltage-zigbee-led-controller-sr-zg1029-5c.html

If I set this within Home Assitant, the color looks almost blue:

```
service: light.turn_on
data:
  rgbw_color:
    - 255
    - 255
    - 255
    - 255
  brightness: 255
```
Results in status:

```
{
    "brightness": 254,
    "color": {
        "h": 0,
        "hue": 0,
        "s": 0,
        "saturation": 0,
        "x": 0.323,
        "y": 0.329
    },
    "color_mode": "xy",
    "color_temp": 168,
    "color_temp_startup": 65535,
    "linkquality": 255,
    "state": "ON"
}

```

I am not sure if this  a bug or something else. It looks like the command from Home Assistant is actually what it resulted, but I need to set RGBW anyway.

```
Received MQTT message on 'zigbee2mqtt/LED Sofa/set' with data '{"state":"ON","color":{"x":0.323,"y":0.329},"brightness":254}'
Publishing 'set' 'brightness' to 'LED Sofa'
Publishing 'set' 'color' to 'LED Sofa'
MQTT publish: topic 'zigbee2mqtt/LED Sofa', payload '{"brightness":254,"color":{"h":0,"hue":0,"s":0,"saturation":0,"x":0.323,"y":0.329},"color_mode":"xy","color_temp":168,"color_temp_startup":65535,"level_config":{"on_level":"previous"},"linkquality":255,"state":"ON"}'
```

**Comment by martinweber** [comment]
I found that the Sunricher SR-ZG1029-5C in RGB+CCT mode is detected as Illuminize 511.000 model as it uses the Zigbee model HK-ZD-RGBCCT-A.

The Sunricher SR-ZG1029-5C has two endpoints in RGB+CCT mode. One for the RGB LEDs and one for the white (warm and cold) LEDs.
The configuration for the Iluminize model only exposes the RGB channel. In my case, I cannot switch or adjust the white leds at all. My controller is a ISOLed 114729 which is a white-labelled Sunricher SR-ZG1029-5C.

I have created a new device configuration for Zigbee model HK-ZD-RGBCCT-A for myself to expose both endpoints. In Home Assistant I see one device with two entities. You need to switch and adjust those two entities (RGB and CCT) separately. I am not sure yet if that is good or bad ;-) 

More details here: https://cgrebel.com/2023/01/zigbee2mqtt-homeassistant-sunricher-sr-zg1029-5c-support/

Here is my config:
```javascript
const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const extend = require('zigbee-herdsman-converters/lib/extend');
const e = exposes.presets;
const ea = exposes.access;

const definition = {
    zigbeeModel: ['HK-ZD-RGBCCT-A'],
    // https://www.sunricher.com/dim-cct-rgbw-rgb-cct-4-in-1-constant-voltage-zigbee-led-controller-sr-zg1029-5c.html
    // The controller can be put into different modes using DIP switches: DIM, CCT, RGBW, and RGB+CCT
    // Also sold as ISOLed Model 114729: https://www.isoled.shop/de/zigbee-30-pwm-dimmer-fuer-led-flexbaenderspots-5-kanal-12-24v-dc-5x4a_114729.html
    model: 'SR-ZG1029-5C',
    vendor: 'Sunricher',
    whiteLabel: [{vendor: 'Sunricher', model: 'SR-ZG1029-5C'}],
    description: 'DIM CCT RGBW RGB+CCT 4-in-1 Constant Voltage Zigbee LED Controller',
    extend: extend.light_onoff_brightness_colortemp_color(),
    exposes: [
        e.light_brightness_colortemp_colorxy().withEndpoint('l1'),
        e.light_brightness_colortemp().withEndpoint('l2')
    ],
    endpoint: (device) => {
        return {'l1' : 1, 'l2' : 2 };
    },
    meta: {multiEndpoint: true},
};


module.exports = definition;
```

**Comment by Ltek** [comment]
@martinweber thanks for sharing, how are these working for you? I want to use them, just cant find anywhere in the USA to buy them yet. thx for the help!

**Reply by martinweber** [reply]
Working ok. Nothing really good or bad to say about the controller.
